src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -5269,3 +5279,103 @@ void MacroAssembler::spin_wait() {
     }
   }
 }
+
+// Stack frame creation/removal
+
+void MacroAssembler::enter() {
+  protect_return_address();
+  stp(rfp, lr, Address(pre(sp, -2 * wordSize)));
+  mov(rfp, sp);
+}
+
+// Create an additional frame for a function.
+void MacroAssembler::enter_subframe() {
+  // Addresses can only be signed once, so strip it first. PAC safe because the value is not
```

Comment:
This needs a more descriptive name. `enter_and_sign()` ?

Reply:
Because enter always enters a subframe. That's what it's for.

Reply:
enter_nested() ?
enter_inner() ?

Reply:
Tell you what, first put a comment here that says when it should (and therefore, should not) be used. Once it's clear exactly what this is for, thinking of a name maight be easier.

Reply:
How about extending the existing enter() function: 

// Enter a new stack frame for the current method.
// nested:     Indicates a frame has already been entered (and not left) for the current method. 
void MacroAssembler::enter(bool nested=false) {
   if (nested) strip()
   protect()
   stp()
   mov()
}

This would add an additional bool check for every call of enter() - that's at code generation time, so probably not an issue.

Reply:
So, `nested` is true iff we are, say, pushing an extra frame for a runtime call in the middle of generated code, but for some mysterious reason the logic is inline instead of being implemented in the obvious way as a stub.

Please do this as:

` MacroAssembler::enter(bool strip_return_address=false)`

and I'll be happy. Please make sure that all calls are commented, as in

`__ enter(/*strip_return_address*/true);`

and I'll be happy.

Reply:
Just about to resolve this ... then spotted the "make sure that all calls are commented". Will fix up.